org.eclipse.jdt.core/search/org/eclipse/jdt/internal/core/search/matching/MatchLocator.java

```diff
@@ -3457,4 +3444,69 @@ protected boolean typeInHierarchy(ReferenceBinding binding) {
 	}
 	return false;
 }
+/*
+ * Go through the nest hierarchy of a type, to try to create a method handle for a method declaration.
+ * Does not try to create a method handle for the specified type, only for types higher in the nest hierarchy.
+ *
+ * See:
+ * https://github.com/eclipse-jdt/eclipse.jdt.core/issues/375
+ * https://bugs.eclipse.org/bugs/show_bug.cgi?id=468127
+ */
+private IJavaElement createHandleFromNestAncestors(AbstractMethodDeclaration method, IJavaElement type) throws JavaModelException {
+	BinaryType binaryType = getBinaryType(type);
+	if (binaryType != null) {
+		boolean isJarOrClass = isJarOrClass(binaryType);
+		if (isJarOrClass) {
+			return findMethodInNestAncestors(method, binaryType);
+		} else {
```

Comment:
I think we can delete this `isJarOrClass()` entirely, if I understand right, we always have "jrt-fs.jar" in modular JDK case => I will create follow up task.

Reply:
#422 created